src/sentry/tasks/summaries/weekly_reports.py

```diff
@@ -211,76 +224,127 @@ def prepare_organization_report(
+                )
+                .values_list("user_id", flat=True)
             )
-            .filter(flags=F("flags").bitand(~OrganizationMember.flags["member-limit:restricted"]))
-            .values_list("user_id", flat=True)
-        )
-        user_list = list(filter(lambda v: v is not None, user_list))
-        user_ids = notifications_service.get_users_for_weekly_reports(
-            organization_id=ctx.organization.id, user_ids=user_list
+            user_list = list(filter(lambda v: v is not None, user_list))
+            user_ids = notifications_service.get_users_for_weekly_reports(
+                organization_id=self.ctx.organization.id, user_ids=user_list
+            )
+            user_template_context_by_user_id_list = []
+            for user_id in user_ids:
+                user_template_context_by_user_id_list = prepare_template_context(
+                    ctx=self.ctx, user_ids=user_ids
+                )
+            if user_template_context_by_user_id_list:
+                for user_template in user_template_context_by_user_id_list:
+                    template_context = user_template.get("context")
+                    user_id = user_template.get("user_id")
+                    if template_context and user_id:
+                        self.send_email(template_ctx=template_context, user_id=user_id)
+
+    def send_email(self, template_ctx: Mapping[str, Any], user_id: int) -> None:
+        message = MessageBuilder(
+            subject=f"Weekly Report for {self.ctx.organization.name}: {date_format(self.ctx.start)} - {date_format(self.ctx.end)}",
+            template="sentry/emails/reports/body.txt",
+            html_template="sentry/emails/reports/body.html",
+            type="report.organization",
+            context=template_ctx,
+            headers={"X-SMTPAPI": json.dumps({"category": "organization_weekly_report"})},
         )
-        user_template_context_by_user_id_list = []
-        for user_id in user_ids:
-            user_template_context_by_user_id_list = prepare_template_context(
-                ctx=ctx, user_ids=user_ids
+        if self.dry_run:
+            return
+
+        if self.email_override:
+            message.send(to=(self.email_override,))
+        else:
+            analytics.record(
+                "weekly_report.sent",
+                user_id=user_id,
+                organization_id=self.ctx.organization.id,
+                notification_uuid=template_ctx["notification_uuid"],
+                user_project_count=template_ctx["user_project_count"],
+            )
+
+            # TODO see if we can use the UUID to track if the email was sent or not
+            logger.info(
+                "weekly_report.send_email",
+                extra={
+                    "batch_id": self.batch_id,
+                    "organization": self.ctx.organization.id,
+                    "uuid": template_ctx["notification_uuid"],
+                },
```

Comment:
Can you add the user id here? When I was looking at these I was wishing we had it